src/components/structures/GroupView.js

```diff
@@ -615,6 +621,11 @@ export default React.createClass({
             });
             dis.dispatch({action: 'panel_disable'});
             this._initGroupStore(this.props.groupId);
+
+            if (this.state.avatarChanged) {
+                // XXX: Evil - poking a store should be done from an async action
+                FlairStore.refreshGroupProfile(this._matrixClient, this.props.groupId);
```

Comment:
hmm - it's not async though? Agree it seems like a hack, but this might just be because the write doesn't go through the store so we have to poke it manually here to tell it to refresh. Either way, it's fine and not worth rearchitecting for now.

Reply:
It is async because we cause the store to do some async work - fetching the group profile. I suppose the source of evil is really the store itself doing the async work. 

Agreed not worth rearchitecting now, let's tackle it when in maintenance mode though for sure.